app/tests/thread_test.cc

```diff
@@ -20,6 +20,17 @@
 #include "gmock/gmock.h"
 #include "gtest/gtest.h"
 
+#define EXPECT_SYSTEM_ERROR(statement, error_code) \
```

Comment:
WDYT about naming this something like EXPECT_THROW_WITH_CODE? (just a thought - so it's clearer in the tests that this is specifically testing for a thrown exception.) Or if you think it isn't too wordy, just to use EXPECT_THROW directly with the EXPECT_EQ inside the catch in each of the tests, rather than the #define, since it's all just in this file anyway.

Reply:
Ah good point. I wanted to avoid repeating the pattern (EXPECT_THROW with EXPECT_EQ)  in multiple places. I could change it to that but it seemed like a lot of code duplication. How about `EXPECT_THROW_WITH_ERROR_CODE`?